lib/neuron/config.ex

```diff
@@ -9,35 +9,65 @@ defmodule Neuron.Config do
       iex> Neuron.Config.set(url: "http://example.com/graph")
       :ok
 
+      iex> Neuron.Config.set(headers: ["name": "val"])
+      :ok
+
       iex> Neuron.Config.set(:process, url: "http://example.com/graph")
       :ok
   """
   def set(value), do: set(:global, value)
-  def set(:global, value) do
-    Application.put_env(:neuron, :graphql, value)
+  def set(:global, nil) do
+    Application.delete_env(:neuron, :neuron_url)
+    Application.delete_env(:neuron, :neuron_headers)
+  end
+
+  def set(:global, url: value) do
+    Application.put_env(:neuron, :neuron_url, value)
+  end
+  def set(:process, url: value) do
+    Process.put(:neuron_url, value)
+    :ok
   end
-  def set(:process, value) do
-    Process.put(:neuron_graphql, value)
+
+  def set(:global, headers: value) do
+    Application.put_env(:neuron, :neuron_headers, value)
+  end
+  def set(:process, headers: value) do
+    Process.put(:neuron_headers, value)
     :ok
   end
 
   @doc """
   gets url configuration value for Neuron
 
-      iex>Neuron.Config.set(url: "http://example.com/graph"); Neuron.Config.get(:url)
+      iex>Neuron.Config.set(url: "http://example.com/graph")
+      iex>Neuron.Config.get(:url)
       "http://example.com/graph"
+
+      iex>Neuron.Config.set(headers: [name: "value"])
+      iex>Neuron.Config.get(:headers)
+      [name: "value"]
+
+      iex>Neuron.Config.get(:invalid)
+      nil
+
   """
-  def get(:url) do
-    case get_config_for(current_context()) do
-      [url: url] -> url
-      _ -> nil
-    end
+  def get(:headers), do: get(:neuron_headers)
+  def get(:url), do: get(:neuron_url)
+  def get(key) do
+    # current_context()
```

Comment:
I think you forgot to remove this code 😁 

Can we have it as something like this?

```elixir
key
|> current_context()
|> get_config_for()
|> Map.get(key)
```